docs/usage/migration.rst

```diff
@@ -0,0 +1,31 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+[WIP] Data Migration
+--------------------
+
+:code:`Locations` are used to represent different storage systems and possibly
+different geographical locations. :code:`Buckets` but also :code:`Objects` are
+assigned a Location. This architecture provides extra flexibility when we want
+to migrate our data.
+
+When we create a bucket we need to provide a Location, otherwise the default
+one is used.
+
+.. note::
+    Before migrating our models, we need to transfer the actual files in the
+    new storage with the new location. Then we can just bulk update our
+    FileInstance objects pointing to the new bucket.
+
+There is a task :code:`invenio_files_rest.tasks.migrate_file`
+which is used to migrate a file instance to a new location. A new location
+might be in remote system on a different bucket, even on a different storage
+backend. If the process does not complete successfully then the destination
+FileInstance is removed completely. The task also accepts an argument
+:code:`post_fixity_check` which enables the file checksum verification after
+the operation is complete.
```

Comment:
Giving my version, feel free to adapt/change it, etc...

`
Invenio-Files-REST provides a Celery task (`invenio_files_rest.tasks.migrate_file`) to migrate a existing files from the current location to a new location. Given a `file_id` and the `name` of the new location (which should have been already created), it will:
1. create a new empty FileInstance in the destination location
2. copy the file content in the newly created FileInstance
3. re-link all ObjectVersions pointing to the previous FileInstance to the new one
4. optionally, re-compute the file checksum

This task can be used to migrate all files or a subset of files in case of location change.
`